R/dt_spanners.R

```diff
@@ -32,15 +34,39 @@ dt_spanners_add <- function(data,
                             vars,
                             spanner_label,
                             spanner_id,
-                            gather) {
+                            spanner_level,
+                            gather,
+                            replace) {
+
+  spanners_at_level <-
+    dt_spanners_get(data = data) %>%
+    dplyr::filter(.env$spanner_level == .data$spanner_level)
+
+  if (
+    !replace &&
+    any(vars %in% unlist(spanners_at_level[["vars"]]))
+  ) {
+
+    error_vars <-
+      paste(
+        vars[vars %in% unlist(spanners_at_level[["vars"]])],
+        collapse = ", "
+      )
 
+    stop(
+      "The column(s) used (", error_vars ,") for the new spanner `",
+      spanner_id, "` belong to an existing spanner",
+      call. = FALSE
+    )
+  }
```

Comment:
If `replace`, make sure to modify existing spanners so they no longer contain that variable.

Reply:
Now done.